sys/net/application_layer/dhcpv6/client.c

```diff
@@ -566,6 +648,52 @@ static void _parse_advertise(uint8_t *adv, size_t len)
                     }
                 }
                 break;
+#if IS_USED(MODULE_GNRC_DHCPV6_CLIENT_IA_NA)
+            case DHCPV6_OPT_IA_NA:
+                for (unsigned i = 0; i < CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX; i++) {
+                    dhcpv6_opt_ia_na_t *ia_na = (dhcpv6_opt_ia_na_t *)opt;
+                    unsigned na_t1, na_t2;
+                    uint32_t ia_id = byteorder_ntohl(ia_na->ia_id);
+                    size_t ia_na_len = byteorder_ntohs(ia_na->len) -
+                                       (sizeof(dhcpv6_opt_ia_na_t) - sizeof(dhcpv6_opt_t));
+                    size_t ia_na_orig_len = ia_na_len;
+
+                    if (addr_leases[i].parent.ia_id.id != ia_id) {
+                        continue;
+                    }
+
+                    /* check for status */
+                    for (dhcpv6_opt_t *ia_na_opt = (dhcpv6_opt_t *)(ia_na + 1);
+                         ia_na_len > 0;
+                         ia_na_len -= _opt_len(ia_na_opt),
+                         ia_na_opt = _opt_next(ia_na_opt)) {
+                        if (ia_na_len > ia_na_orig_len) {
+                            DEBUG("DHCPv6 client: IA_NA options overflow option "
+                                  "boundaries\n");
+                            return;
+                        }
+                        switch (byteorder_ntohs(ia_na_opt->type)) {
+                            case DHCPV6_OPT_STATUS: {
+                                if (!_check_status_opt((dhcpv6_opt_status_t *)ia_na_opt)) {
+                                    continue;
+                                }
+                                break;
+                            }
+                            default:
+                                break;
+                        }
+                    }
+                    na_t1 = byteorder_ntohl(ia_na->t1);
+                    na_t2 = byteorder_ntohl(ia_na->t2);
+                    if ((na_t1 != 0) && (na_t2 != 0) &&
+                        (server.t1 > na_t1) && (t2 > na_t2)) {
+                        server.t1 = na_t1;
+                        t2 = na_t2;
+                        _schedule_t2();
+                    }
+                }
+                break;
```

Comment:
This whole block looks somewhat complex. Maybe put it in its own function?

Reply:
I now created a separate function called `_parse_ia_na_option` for this code block.